File: src/main/JavaScript/Vigenere Cipher/Vigener Cipher.ts
//Here i take alphabet veriable for to convert letter to numerical values
//Vigenère Cipher
class VigenerCipher {
  ALPHABET = " ABCDEFGHIJKLMNOPQRSTUVWXYZ";
  constructor() {}
  public vigenerEncrypt(message: string, key: string): string {
    //This is the text that we want to encrypt
    message = message.toUpperCase();
    key = key.toUpperCase();
    let cipher_text = " ";
    //here i take key index is zero initially to represent the key in numerical values
    let key_index = 0;
    let index = 0;
    for (const i of message) {
      index = (this.ALPHABET.indexOf(i) + this.ALPHABET.indexOf(key[key_index])) % this.ALPHABET.length;
      //   console.log(index);
      cipher_text += this.ALPHABET.charAt(index);
      key_index++;
      if (key_index === key.length) key_index = 0;

    }
    return cipher_text;
  }
  public vigenerDecrypt(message: string, key: string): string {
    //This is the text that we want to encrypt
    message = message.toUpperCase();
    key = key.toUpperCase();
    let cipher_text = " ";
    //here i take key index is zero initially to represent the key in numerical values
    let key_index = 0;
    let index = 0;
    for (const i of message) {
      index = (this.ALPHABET.indexOf(i) - this.ALPHABET.indexOf(key[key_index])) % this.ALPHABET.length;
      //   console.log(index);
      cipher_text += this.ALPHABET.charAt(index);
      key_index++;
      if (key_index === key.length) key_index = 0;

    }
    return cipher_text;
  }

}
//Testing Perpose
let obj = new VigenerCipher();
//console.log(obj.vigenerEncrypt("this is just an example", "secret"));
let sms = obj.vigenerEncrypt("Bangladesh", "secret");
 console.info(sms);
let msm = obj.vigenerDecrypt("UFQYQUWJVZ", "secret");
 console.info(msm);

File: src/main/python/VigenerCipher/Vigener Cipher.py
ALBHABET=" ABCDEFGHIJKLMNOPQRSTUVWXYZ"
class VigenerCipher:
    def __init__(self) -> None:
        pass
    def Encrypt(message,key):
        #this is the letter that we want to encrypt+in case sensative
        message=message.upper()
        key=key.upper()
        cipher_text=' '
        key_index=0;
        for char in message:
            index=(ALBHABET.find(char)+ALBHABET.find(key[key_index]))%len(ALBHABET)
            cipher_text+=ALBHABET[index];
            key_index+=0
            if key_index==len(key):
                key_index=0;
        return cipher_text
    def Decrypt(message,key):
        #this is the letter that we want to encrypt+in case sensative
        message=message.upper()
        key=key.upper()
        cipher_text=' '
        key_index=0;
        for char in message:
            index=(ALBHABET.find(char)-ALBHABET.find(key[key_index]))%len(ALBHABET)
            cipher_text+=ALBHABET[index];
            key_index+=0
            if key_index==len(key):
                key_index=0;
        return cipher_text
if __name__=='__main__':
    print(VigenerCipher.Encrypt("I LOVE YOU","SECRET"))
    print(VigenerCipher.Decrypt("ASDGNXSQGM","SECRET"))
